packages/frontend/app/components/HeaderBar.tsx

```diff
@@ -408,14 +412,46 @@ const HeaderBar: React.FC = () => {
         unoptimized
       />
       <div className="flex h-14 grow flex-col items-center justify-center">
-        <h1 className="leading-none">
+        <h1
+          className={cn(
+            'leading-none',
+            !course?.name
+              ? ''
+              : course.name.length > 35
+                ? 'text-xs'
+                : course.name.length > 30
+                  ? 'text-sm'
+                  : course.name.length > 25
+                    ? 'text-base'
+                    : course.name.length > 20
+                      ? 'text-lg'
+                      : course.name.length > 15
+                        ? 'text-xl'
+                        : '',
+          )}
+        >
           {isProfilePage ? 'Profile' : course?.name}
         </h1>
-        <h2 className="text-base leading-none text-slate-500">
-          {queueId
-            ? course?.queues?.find((queue) => queue.id === queueId)?.room
-            : ''}
-        </h2>
+        {queueRoom && (
+          <h2
+            className={cn(
+              'text-base leading-none text-slate-500',
+              queueRoom.length > 35
+                ? 'text-xs'
+                : queueRoom.length > 30
+                  ? 'text-sm'
+                  : queueRoom.length > 25
+                    ? 'text-base'
+                    : queueRoom.length > 20
+                      ? 'text-lg'
+                      : queueRoom.length > 15
+                        ? 'text-xl'
+                        : '',
+            )}
+          >
+            {queueRoom}
+          </h2>
+        )}
```

Comment:
>  text size should not change based on length, ways to deal can be truncation, wrapping, or scrollables

In regards to this, generally this is a correct sentiment. However, since this is the title in the header bar, truncating, wrapping, and scrolling all don't work particularly well.

> Second is that if this is done, I don't mind ternary operators, but the two should be combined into a reusable function.

Man none of y'all would have batted an eye if prettier didn't expand my ternary operators to the moon 😭 
I wanted to keep them separate since that way they aren't tied to one another... and speaking of, i realised that my current code increases the size of the queue text since text-xl is larger than its original size, so i'll go fix that